pymeasure/instruments/agilent/agilentN9320A.py

```diff
@@ -0,0 +1,296 @@
+        in seconds. This property can be set."""
+    )
+    ref_level = Instrument.control(
+        ":DISP:WIND:TRAC:Y:SCAL:RLEV?", ":DISP:WIND:TRAC:Y:SCAL:RLEV %.2f",
+        """ A floating point property that represents the reference level
+        in dBm. This property can be set.""",
+        validator=strict_range,
+        values=REF_LIMIT
+    )
+    attenuation = Instrument.control(
+        ":SENS:POW:RF:ATT?", ":SENS:POW:RF:ATT %d",
+        """ A floating point property that represents the input attenuator
+        in dB. This property can be set.""",
+        validator=strict_range,
+        values=ATT_LIMIT
+    )
+    unit = Instrument.control(
+        "UNIT:POW?", "UNIT:POW %s",
+        """ A floating point property that represents the amplitude unit.
+        This property can be set.""",
+        validator=strict_discrete_set,
+        values=UNITS
+    )
+    resolution_bandwidth = Instrument.control(
+        ":SENS:BAND:RES?", ":SENS:BAND:RES %e",
+        """ A floating point property that represents the resolution
+        bandwidth in Hz. This property can be set.""",
+        validator=strict_discrete_set,
+        values=RES_LIMIT
+    )
+    video_bandwidth = Instrument.control(
+        ":SENS:BAND:VID?", ":SENS:BAND:VID %e",
+        """ A floating point property that represents the video
+        bandwidth in Hz. This property can be set.""",
+        validator=strict_discrete_set,
+        values=VID_LIMIT
+    )
+
+    def __init__(self, adapter, **kwargs):
+        super(AgilentN9320A, self).__init__(
+            adapter,
+            "Agilent AgilentN9320A Spectrum Analyzer",
+            **kwargs
+        )
+
+    def display_on(self):
+        """ Switch on the display. """
+        self.write("DISP:ENAB 1")
+
+    def display_off(self):
+        """ Switch off the display. """
+        self.write("DISP:ENAB 0")
+
+    def init_immediate(self):
+        """ This method initiates a sweep if not in a measurement. If
+        in a measurement, it triggers the measurement. """
+        self.write("INIT:IMM")
+
+    def init_continuos(self):
+        """ Trigger system is continuosly initiated. """
+        self.write("INIT:CONT 1")
+
+    def init_single(self):
+        """ The sweep system remains in an idle state until init_continuos
+        is set or init_immediate is received. When init_immediate
+        command received, it will go through a single sweep cycle, and then
+        return to the idle state. """
+        self.write("INIT:CONT 0")
+
+    def set_opc_sqr(self):
+        """ Set the instrument to generate SRQ when operation is complete:
+        remember to run a *CLS before the command that you want
+        to sync/wait for. Timeout in seconds. """
+        self.write("*ESE 1;*SRE 32")
+
+    def opc(self):
+        """ It returns 1 when operation complete. You may need to change the
+        visa timeout (in ms) for long lasting operations. """
+        self.ask("*OPC?")
```

Comment:
This function always returns `None`, irrespective of the result of the ask() mentioned in the docstring. How do you tell the difference between a successful and an aborted wait?
I don't know how the device handles long operations, but it might be useful to include custom timeout handling here instead of below in peak_output. you can look how other instruments in pymeasure deal with `OPC` instructions.